### ISSUE TITLE
need php mysql example for saving and updating page content 

### ISSUE THREAD

**Issue by umefarooq** [Kademi/keditor#33]
How to save changes of keditor in database with php mysql, looking for a good example.

**Comment by bradmac** [comment]
Here's what we do in Kademi

```
    btnSaveFile.on('click', function (e) {
        e.preventDefault();

        showLoadingIcon();
        var fileContent = $('#content-area').keditor('getContent');
        var saveUrl = postMessageData && postMessageData.pageName ? postMessageData.pageName : fileName;

        $.ajax({
            url: saveUrl,
            type: 'POST',
            data: {
                body: fileContent
            },            
            dataType: 'json',
            success: function () {
                if (isEmbeddedIframe) {
                    doPostMessage({
                        isSaved: true,
                        resp: postMessageData.resp,
                        willClose: postMessageData.willClose
                    }, postMessageData.url);
                } else {
                    Msg.success('File is saved!');
                }

                hideLoadingIcon();
                body.removeClass('content-changed');
            },
            error: function (e) {
                Msg.error(e.status + ': ' + e.statusText);
                hideLoadingIcon();
            }
        })
    });

```

**Comment by umefarooq** [comment]
so quick let me try if its work really thanks.

**Comment by umefarooq** [comment]
it worked amazing, now part of my development list, how can i disable all console logs.

**Comment by anhnt** [comment]
@umefarooq can you create an issue for it? I would assume that it should be an option like 
``` $('div').keditor({debug: false});```

**Comment by bradmac** [comment]
@umefarooq note that logging uses a function which checks to see if the console is enabled, so it is safe to leave logging enabled in production.

**Comment by ducdhm** [comment]
@umefarooq: If you want, you still can disable log console by using this code `$.keditor.debug = false;`

**Comment by umarch06** [comment]
Amazing plugin.
@ducdhm @bradmac  can you please tell me if I can again load the page design after saving html through this line of code?

`var fileContent = $('#keditor-body').keditor('getContent');`

**Comment by bradmac** [comment]
Hey @umarch06 what do you mean by loading the page design again? In Kademi we save the content with ajax and the user just continues editing, so the page is not reloaded.

**Comment by umarch06** [comment]
@bradmac consider that I need to save a page and then I have to reload it with the same design I left with.
Like in an invoice, a person may have as much design as he want and later on he needs to build a new invoice with some of his custom designs?

**Comment by bradmac** [comment]
OK, so it sounds like what you want is to save to a template library, and allow your users to load from that template library, right?

You can definitely do that, but i would expect thats outside of the scope of keditor. You would want to create a new table in your db, create an ajax service to get and set templates, and then have some sort of drop down to select one which would load the content into the content editor

**Comment by umarch06** [comment]
yes exactly @bradmac you got me right.
I need to save templates. Later on I need to use them again, even again I can have right to make changes in that.
The question is relative. Like thorugh this code `var fileContent = $('#keditor-body').keditor('getContent');`
I am gonna get all the html. Later on I will save it to db and then again I will reload it on my page. 
Problem is that, how will I render that again? How would all the stuff be created again, like containers, snippets and anything else. Please refer to this url http://angular-invoice-builder.azurewebsites.net
How would be everything get back to that state which I was before saving and reloading (like elements on invoice page, the draggers and each and evey property like drag, settings etc would be needed to remain same after reloading)?
I have created a new tab that names **Invoice Items**. Keep in mind that it was not created by using ExtraTabs in your plugin. I myself created it through some custom logic. Since it needs to be dynamic.

**Note** that I will remove this url which I provided above after your comment.

**Comment by bradmac** [comment]
We generate editor page like this (uses velocity):
```
<div id="content-area">$!file.body</div>
```

And then we initialise it like this:
```
    contentArea.keditor({
        ckeditorOptions: {
            skin: editorSkin,
...
```

To load new content we would simple replace the content div and then re-init keditor
```
$("#content-area").html(newContent);
contentArea.keditor({..});
```

The content produced by keditor should be clean, it should not contain anything editor specific such as draggers.

**Comment by bradmac** [comment]
BTW, you should really consider using dynamic components. Check it out here, but you might want to skip my long boring intro and skip forward to about 4:30 :)

http://www.kademi.co/blogs/kb/building-customized-analytics-dashboards/

**Comment by umarch06** [comment]
@bradmac I loved the way you have implemented this. I watched your video till the end and wonder why in that case Elements and Containers are on same tab and some other snippets are separately placed on second tab of side bar.
- Moreover, it would have been excellent if one could drag elements on content area rather than first placing containers and then placing elements in those containers (most of us need to drag single  element) to some custom space.
- I would also need to suggest you a thing that the containers should be move able in their parent content areas like I did in the (deleted)URL I provided above.  Resizing is a plus too. The reason I am asking for so is that almost every drag and drop I have seen on internet restricts user to push the containers in stack (containers are not free to move or in simple words I can say I cannot place the element exactly at that position that I wish).
- Coming to dynamic content, what If I need the things to come from some webservices like in my case and than show them in side bar. Since I can see the html has been restricted to show the elements only if they are present in dyamic_content file.
